### PR TITLE
chore!: Remove `InterfaceApi` in favor of `InterfaceMessagesApi`

### PR DESCRIPTION
**Pull request by kulikthebird**

### CODE REVIEW

**Review by jawoznia — CHANGES_REQUESTED** [review]
Nice, but please mark this change as breaking in commit message

File: sylvia-derive/src/interface/communication/api.rs [hunk 1]
```diff
@@ -58,45 +49,27 @@ impl<'a> Api<'a> {
...
-                type Querier<'querier> = #sylvia ::types::BoundQuerier<'querier, #custom_query, Contract >;
-            }
-
-
-            pub struct Api < #(#generics,)* > {
-                #phantom
             }
 
-            impl < #(#generics,)* > #sylvia ::types::InterfaceApi for Api < #(#generics,)* > #where_clause {
+            impl<'a, Error, #(#generics),*> InterfaceMessagesApi for dyn #interface_name < Error = Error, #(#generics = #generics,)* > + 'a {
```
Comment (jawoznia) [inline sylvia-derive/src/interface/communication/api.rs:69]: It should not be an issue here, but please keep the convention of naming generated lifetimes and make it more unique with `sv_` prefix

**Review by jawoznia — APPROVED** [review]
One comment. Otherwise LGTM

File: sylvia-derive/src/interface/communication/api.rs (hunk 1 above)
Comment (kulikthebird) [reply sylvia-derive/src/interface/communication/api.rs]: Done

**Pull request merged**